src/set_theory/ordinal_arithmetic.lean

```diff
@@ -2019,7 +2019,7 @@ by rw [CNF_rec, dif_neg o0]
     base-`b` expansion of `o`.
 
     `CNF b (b ^ u₁ * v₁ + b ^ u₂ * v₂) = [(u₁, v₁), (u₂, v₂)]` -/
-def CNF (b := omega) (o : ordinal) : list (ordinal × ordinal) :=
+def CNF (b o : ordinal) : list (ordinal × ordinal) :=
```

Comment:
If you require that `b` and `o` have the same universe level in almost every property about `CNF`, shouldn't you just require it in the definition of `CNF`?

Reply:
It is required in the definition. But writing `p ∈ ordinal × ordinal` seems to confuse the kernel, and it ends with all of my lemmas having everything in universe `max ?1 ?2` instead of using a single universe. This is why I need the universe annotations.

Reply:
Oh I see. Yeah, that sometimes happens. Thanks for the clarification!